src/test/app/Batch_test.cpp

```diff
@@ -0,0 +1,1661 @@
+            batch::msig(bob, {dave, carol}));
+        env.close();
+
+        std::vector<TestBatchData> testCases = {
+            {"tesSUCCESS", "Payment"},
+            {"tesSUCCESS", "Payment"},
+        };
+
+        Json::Value params;
+        params[jss::ledger_index] = env.current()->seq() - 1;
+        params[jss::transactions] = true;
+        params[jss::expand] = true;
+        auto const jrr = env.rpc("json", "ledger", to_string(params));
+        auto const txn = getTxByIndex(jrr, 2);
+        validateBatchTxns(txn[jss::metaData], 3, testCases);
+        validateBatchMeta(txn[jss::metaData], STAmount(XRP(1000)), 4);
+
+        BEAST_EXPECT(env.seq(alice) == 6);
+        BEAST_EXPECT(env.seq(bob) == 6);
+        BEAST_EXPECT(env.balance(alice) == preAlice - XRP(5) - batchFee);
+        BEAST_EXPECT(env.balance(bob) == preBob + XRP(5));
+    }
+
+    void
+    testSubmit(FeatureBitset features)
+    {
+        testcase("submit");
+
+        using namespace test::jtx;
+        using namespace std::literals;
+
+        test::jtx::Env env{*this, envconfig()};
+
+        auto const alice = Account("alice");
+        auto const bob = Account("bob");
+        auto const gw = Account("gw");
+        auto const USD = gw["USD"];
+
+        env.fund(XRP(1000), alice, bob, gw);
+        env.close();
+
+        env.trust(USD(1000), alice, bob);
+        env(pay(gw, alice, USD(100)));
+        env(pay(gw, bob, USD(100)));
+        env.close();
+
+        // Invalid: txn has `tfInnerBatchTxn` flag and signature
+        {
+            auto jv = pay(alice, bob, USD(1));
+            jv[sfFlags.fieldName] = tfInnerBatchTxn;
+
+            Serializer s;
+            auto jt = env.jt(jv);
+            jv.removeMember(sfTxnSignature.jsonName);
+            s.erase();
+            jt.stx->add(s);
+            auto const jrr = env.rpc("submit", strHex(s.slice()))[jss::result];
+            BEAST_EXPECT(
+                jrr[jss::status] == "error" &&
+                jrr[jss::error] == "invalidTransaction");
+
+            env.close();
+        }
+
+        // Invalid: txn has `tfInnerBatchTxn` flag and no signature
+        {
+            std::string txBlob =
+                "1200002240000000240000000561D4838D7EA4C68000000000000000000000"
+                "0000005553440000000000A407AF5856CCF3C42619DAA925813FC955C72983"
+                "68400000000000000A73210388935426E0D08083314842EDFBB2D517BD4769"
+                "9F9A4527318A8E10468C97C0528114AE123A8556F3CF91154711376AFB0F89"
+                "4F832B3D8314F51DFC2A09D62CBBA1DFBDD4691DAC96AD98B90F";
+            auto const jrr = env.rpc("submit", txBlob)[jss::result];
+            BEAST_EXPECT(
+                jrr[jss::status] == "success" &&
+                jrr[jss::engine_result] == "temINVALID_BATCH");
+
+            env.close();
+        }
+    }
+
+    void
+    testNoAccount(FeatureBitset features)
+    {
+        testcase("no account");
```

Comment:
is this the right name for the test? seems like the test is just submitting a batch with payment and accountset. there is a purpose for testing this combo specifically?

Reply:
This test is submitting a payment to an account that does not exist and then doing the AccountSet. Its just there to test that account activation works. I can change the name? What name would you suggest? Or should I remove the test all together?

Reply:
I see that makes sense, might be worth aadding a comment like 
```
// Create Bob's account by funding it with XRP as an inner transaction.  
// Then Bob submits an AccountSet transaction as another inner transaction.  
```